api/src/main/java/net/kyori/adventure/text/format/gradient/HueGradient.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * This file is part of adventure, licensed under the MIT License.
+ *
+ * Copyright (c) 2017-2023 KyoriPowered
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package net.kyori.adventure.text.format.gradient;
+
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * A HueGradient is a gradient whose {@code ColorSpace} involves some form of hue coordinate, which needs
+ * to be specially interpolated.
+ *
+ * @see <a href="https://www.w3.org/TR/css-color-4/#hue-interpolation">W3C CSS Color Level 4 spec on hue interpolation</a>
+ *
+ * @since 4.15.0
+ */
+public abstract class HueGradient implements Gradient {
```

Comment:
All API types should be `interface` - implementations should be package-private.